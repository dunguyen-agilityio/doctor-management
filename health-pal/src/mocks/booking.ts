import { BookingData } from '@/models/booking'

import { BOOKING_TABS } from '@/types/booking'

import { MOCK_DOCTORS } from './doctor'
import { MOCK_PATIENT } from './patient'

export const MOCK_BOOKING: BookingData = {
  id: 1,
  documentId: 'doc-1',
  date: '2025-06-01',
  time: '09:00 AM',
  type: BOOKING_TABS.UPCOMING,
  patient: MOCK_PATIENT,
  doctor: MOCK_DOCTORS[0],
}
