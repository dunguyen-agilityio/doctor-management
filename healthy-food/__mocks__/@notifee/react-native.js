export default {
  requestPermission: jest.fn(() => Promise.resolve(true)),
  displayNotification: jest.fn(() => Promise.resolve()),
  onForegroundEvent: jest.fn(),
  onBackgroundEvent: jest.fn(),
  createChannel: jest.fn(() => Promise.resolve('mock-channel')),
  getInitialNotification: jest.fn(() => Promise.resolve(null)),
};
