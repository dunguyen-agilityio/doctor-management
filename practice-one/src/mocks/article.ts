import { IArticle } from '@/types';

export const MOCK_ARTICLES: IArticle[] = [
  {
    id: '1',
    title: 'The pros and cons of vegetable',
    name: 'Article',
    color: 'green',
    backgroundColor: ['#b3f6a8', '#ebfff3'],
    image:
      'https://res.cloudinary.com/dn5vw6fjp/image/upload/v1740114016/24d2e37a7dcbf108d35e8be410c96a58_uifiia.png',
  },
  {
    id: '2',
    title: 'The pros and cons of fast food.',
    color: 'secondary',
    name: 'Article',
    backgroundColor: ['#fff2f0', '#fff8eb'],
    image:
      'https://res.cloudinary.com/dn5vw6fjp/image/upload/v1738897004/Group_1_q9kcho.png',
  },
  {
    id: '3',
    title: 'The pros and cons of vegetable',
    name: 'Article',
    color: 'green',
    backgroundColor: ['#b3f6a8', '#ebfff3'],
    image:
      'https://res.cloudinary.com/dn5vw6fjp/image/upload/v1740114016/24d2e37a7dcbf108d35e8be410c96a58_uifiia.png',
  },
];
