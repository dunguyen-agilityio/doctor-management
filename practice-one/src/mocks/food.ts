import { IFood } from '@/types';

export const MOCK_FOOD_LIST: IFood[] = [
  {
    id: '22',
    name: 'Pizza',
    weight: 200,
    ingredients: [
      {
        id: '1',
        name: 'Bread',
        value: 200,
      },
      {
        id: '2',
        name: 'Meat (Chicken)',
        value: 500,
      },
      {
        id: '3',
        name: 'Cucumber',
        value: 300,
      },
      {
        id: '4',
        name: 'Onion',
        value: 200,
      },
    ],
    nutritional: {
      calories: 100,
      carbs: 100,
      protein: 100,
      fat: 100,
    },
    category: '1',
    color: 'RED',
    imgUrl:
      'https://s3-alpha-sig.figma.com/img/0d17/c4d2/3b2ee1931e0c9de420fcbd861f82a29a?Expires=1739750400&Key-Pair-Id=APKAQ4GOSFWCW27IBOMQ&Signature=Kk7cHm44D-dhlRtVx3PsLJimx9ElGp1B1ZVxjk~020IbqEEYgd83GQp-7--731jfo0qy4U42pvGksvbN5HjY6OQO8iei44zogdIMdsFzfmaXFwuhNsjH4dZvZ0u0UTPrxiNPVcy969pmpPKjehOI7wPI4iCIaFeZVgMFbLuH0h4b-P0RAdrpZVKE9--rAS540cWY71ix8OHIENnsj~91X8wuOfoZh1NNgs1rWRZew8PMO3y8vEsyyD62OX6oYAfDVEA83TkVKq7Xxti8mVQMx7-A-Cs6T4h1L1QLE0cPUXRGwV-C3DjCTXt5zI0vlGIzWh~hPbGzH57VtGhqWDrX0Q__',
    desc: "Note that in TypeScript, you can define the return type of the function using a colon (:) after the function parameter list. In this example, we don't explicitly define the return type, but TypeScript will infer that the return type is a Promise that resolves to a JSON object.",
    favorite: false,
  },
  {
    id: '1',
    name: 'Chicken 1',
    weight: 400,
    color: 'RED',
    desc: "Note that in TypeScript, you can define the return type of the function using a colon after the function parameter list. In this example, we don't explicitly define the return type, but TypeScript will infer that the return type is a Promise that resolves to a JSON object.",
    nutritional: {
      calories: 200,
      carbs: 100,
      fat: 100,
      protein: 100,
    },
    category: '7',
    ingredients: [
      {
        id: '1',
        name: 'Bread',
        value: 200,
      },
      {
        id: '2',
        name: 'Meat (Chicken)',
        value: 500,
      },
      {
        id: '3',
        name: 'Cucumber',
        value: 300,
      },
      {
        id: '4',
        name: 'Onion',
        value: 200,
      },
    ],
    imgUrl:
      'https://s3-alpha-sig.figma.com/img/0d17/c4d2/3b2ee1931e0c9de420fcbd861f82a29a?Expires=1739750400&Key-Pair-Id=APKAQ4GOSFWCW27IBOMQ&Signature=Kk7cHm44D-dhlRtVx3PsLJimx9ElGp1B1ZVxjk~020IbqEEYgd83GQp-7--731jfo0qy4U42pvGksvbN5HjY6OQO8iei44zogdIMdsFzfmaXFwuhNsjH4dZvZ0u0UTPrxiNPVcy969pmpPKjehOI7wPI4iCIaFeZVgMFbLuH0h4b-P0RAdrpZVKE9--rAS540cWY71ix8OHIENnsj~91X8wuOfoZh1NNgs1rWRZew8PMO3y8vEsyyD62OX6oYAfDVEA83TkVKq7Xxti8mVQMx7-A-Cs6T4h1L1QLE0cPUXRGwV-C3DjCTXt5zI0vlGIzWh~hPbGzH57VtGhqWDrX0Q__',
    favorite: true,
  },
  {
    id: '4',
    desc: "Note that in TypeScript, you can define the return type of the function using a colon (:) after the function parameter list. In this example, we don't explicitly define the return type, but TypeScript will infer that the return type is a Promise that resolves to a JSON object.",
    name: 'Fish',
    weight: 400,
    color: 'YELLOW',
    nutritional: {
      calories: 200,
      carbs: 100,
      fat: 100,
      protein: 100,
    },
    category: '3',
    ingredients: [
      {
        id: '1',
        name: 'Bread',
        value: 200,
      },
      {
        id: '2',
        name: 'Meat (Chicken)',
        value: 500,
      },
      {
        id: '3',
        name: 'Cucumber',
        value: 300,
      },
      {
        id: '4',
        name: 'Onion',
        value: 200,
      },
    ],
    imgUrl:
      'https://s3-alpha-sig.figma.com/img/0d17/c4d2/3b2ee1931e0c9de420fcbd861f82a29a?Expires=1739750400&Key-Pair-Id=APKAQ4GOSFWCW27IBOMQ&Signature=Kk7cHm44D-dhlRtVx3PsLJimx9ElGp1B1ZVxjk~020IbqEEYgd83GQp-7--731jfo0qy4U42pvGksvbN5HjY6OQO8iei44zogdIMdsFzfmaXFwuhNsjH4dZvZ0u0UTPrxiNPVcy969pmpPKjehOI7wPI4iCIaFeZVgMFbLuH0h4b-P0RAdrpZVKE9--rAS540cWY71ix8OHIENnsj~91X8wuOfoZh1NNgs1rWRZew8PMO3y8vEsyyD62OX6oYAfDVEA83TkVKq7Xxti8mVQMx7-A-Cs6T4h1L1QLE0cPUXRGwV-C3DjCTXt5zI0vlGIzWh~hPbGzH57VtGhqWDrX0Q__',
    favorite: false,
  },
  {
    id: '5',
    name: 'Bacon',
    weight: 400,
    color: 'PURPLE',
    desc: "Note that in TypeScript, you can define the return type of the function using a colon (:) after the function parameter list. In this example, we don't explicitly define the return type, but TypeScript will infer that the return type is a Promise that resolves to a JSON object.",
    nutritional: {
      calories: 200,
      carbs: 100,
      fat: 100,
      protein: 100,
    },
    category: '3',
    ingredients: [
      {
        id: '1',
        name: 'Bread',
        value: 200,
      },
      {
        id: '2',
        name: 'Meat (Chicken)',
        value: 500,
      },
      {
        id: '3',
        name: 'Cucumber',
        value: 300,
      },
      {
        id: '4',
        name: 'Onion',
        value: 200,
      },
    ],
    imgUrl:
      'https://s3-alpha-sig.figma.com/img/0d17/c4d2/3b2ee1931e0c9de420fcbd861f82a29a?Expires=1739750400&Key-Pair-Id=APKAQ4GOSFWCW27IBOMQ&Signature=Kk7cHm44D-dhlRtVx3PsLJimx9ElGp1B1ZVxjk~020IbqEEYgd83GQp-7--731jfo0qy4U42pvGksvbN5HjY6OQO8iei44zogdIMdsFzfmaXFwuhNsjH4dZvZ0u0UTPrxiNPVcy969pmpPKjehOI7wPI4iCIaFeZVgMFbLuH0h4b-P0RAdrpZVKE9--rAS540cWY71ix8OHIENnsj~91X8wuOfoZh1NNgs1rWRZew8PMO3y8vEsyyD62OX6oYAfDVEA83TkVKq7Xxti8mVQMx7-A-Cs6T4h1L1QLE0cPUXRGwV-C3DjCTXt5zI0vlGIzWh~hPbGzH57VtGhqWDrX0Q__',
    favorite: true,
  },
  {
    id: '6',
    name: 'Chicken 2',
    weight: 400,
    color: 'RED',
    desc: "Note that in TypeScript, you can define the return type of the function using a colon (:) after the function parameter list. In this example, we don't explicitly define the return type, but TypeScript will infer that the return type is a Promise that resolves to a JSON object.",
    nutritional: {
      calories: 200,
      carbs: 100,
      fat: 100,
      protein: 100,
    },
    favorite: false,
    category: '2',
    ingredients: [
      {
        id: '1',
        name: 'Bread',
        value: 200,
      },
      {
        id: '2',
        name: 'Meat (Chicken)',
        value: 500,
      },
      {
        id: '3',
        name: 'Cucumber',
        value: 300,
      },
      {
        id: '4',
        name: 'Onion',
        value: 200,
      },
    ],
    imgUrl:
      'https://s3-alpha-sig.figma.com/img/0d17/c4d2/3b2ee1931e0c9de420fcbd861f82a29a?Expires=1739750400&Key-Pair-Id=APKAQ4GOSFWCW27IBOMQ&Signature=Kk7cHm44D-dhlRtVx3PsLJimx9ElGp1B1ZVxjk~020IbqEEYgd83GQp-7--731jfo0qy4U42pvGksvbN5HjY6OQO8iei44zogdIMdsFzfmaXFwuhNsjH4dZvZ0u0UTPrxiNPVcy969pmpPKjehOI7wPI4iCIaFeZVgMFbLuH0h4b-P0RAdrpZVKE9--rAS540cWY71ix8OHIENnsj~91X8wuOfoZh1NNgs1rWRZew8PMO3y8vEsyyD62OX6oYAfDVEA83TkVKq7Xxti8mVQMx7-A-Cs6T4h1L1QLE0cPUXRGwV-C3DjCTXt5zI0vlGIzWh~hPbGzH57VtGhqWDrX0Q__',
  },
  {
    id: '7',
    desc: "Note that in TypeScript, you can define the return type of the function using a colon (:) after the function parameter list. In this example, we don't explicitly define the return type, but TypeScript will infer that the return type is a Promise that resolves to a JSON object.",
    name: 'Beef',
    weight: 400,
    color: 'GREEN',
    nutritional: {
      calories: 200,
      carbs: 100,
      fat: 100,
      protein: 100,
    },
    category: '2',
    ingredients: [
      {
        id: '1',
        name: 'Bread',
        value: 200,
      },
      {
        id: '2',
        name: 'Meat (Chicken)',
        value: 500,
      },
      {
        id: '3',
        name: 'Cucumber',
        value: 300,
      },
      {
        id: '4',
        name: 'Onion',
        value: 200,
      },
    ],
    imgUrl:
      'https://s3-alpha-sig.figma.com/img/0d17/c4d2/3b2ee1931e0c9de420fcbd861f82a29a?Expires=1739750400&Key-Pair-Id=APKAQ4GOSFWCW27IBOMQ&Signature=Kk7cHm44D-dhlRtVx3PsLJimx9ElGp1B1ZVxjk~020IbqEEYgd83GQp-7--731jfo0qy4U42pvGksvbN5HjY6OQO8iei44zogdIMdsFzfmaXFwuhNsjH4dZvZ0u0UTPrxiNPVcy969pmpPKjehOI7wPI4iCIaFeZVgMFbLuH0h4b-P0RAdrpZVKE9--rAS540cWY71ix8OHIENnsj~91X8wuOfoZh1NNgs1rWRZew8PMO3y8vEsyyD62OX6oYAfDVEA83TkVKq7Xxti8mVQMx7-A-Cs6T4h1L1QLE0cPUXRGwV-C3DjCTXt5zI0vlGIzWh~hPbGzH57VtGhqWDrX0Q__',
    favorite: false,
  },
  {
    id: '8',
    desc: "Note that in TypeScript, you can define the return type of the function using a colon (:) after the function parameter list. In this example, we don't explicitly define the return type, but TypeScript will infer that the return type is a Promise that resolves to a JSON object.",
    name: 'Pork',
    weight: 400,
    color: 'GREEN',
    nutritional: {
      calories: 200,
      carbs: 100,
      fat: 100,
      protein: 100,
    },
    category: '3',
    ingredients: [
      {
        id: '1',
        name: 'Bread',
        value: 200,
      },
      {
        id: '2',
        name: 'Meat (Chicken)',
        value: 500,
      },
      {
        id: '3',
        name: 'Cucumber',
        value: 300,
      },
      {
        id: '4',
        name: 'Onion',
        value: 200,
      },
    ],
    imgUrl:
      'https://s3-alpha-sig.figma.com/img/0d17/c4d2/3b2ee1931e0c9de420fcbd861f82a29a?Expires=1739750400&Key-Pair-Id=APKAQ4GOSFWCW27IBOMQ&Signature=Kk7cHm44D-dhlRtVx3PsLJimx9ElGp1B1ZVxjk~020IbqEEYgd83GQp-7--731jfo0qy4U42pvGksvbN5HjY6OQO8iei44zogdIMdsFzfmaXFwuhNsjH4dZvZ0u0UTPrxiNPVcy969pmpPKjehOI7wPI4iCIaFeZVgMFbLuH0h4b-P0RAdrpZVKE9--rAS540cWY71ix8OHIENnsj~91X8wuOfoZh1NNgs1rWRZew8PMO3y8vEsyyD62OX6oYAfDVEA83TkVKq7Xxti8mVQMx7-A-Cs6T4h1L1QLE0cPUXRGwV-C3DjCTXt5zI0vlGIzWh~hPbGzH57VtGhqWDrX0Q__',
    favorite: false,
  },
  {
    id: '9',
    desc: "Note that in TypeScript, you can define the return type of the function using a colon (:) after the function parameter list. In this example, we don't explicitly define the return type, but TypeScript will infer that the return type is a Promise that resolves to a JSON object.",
    name: 'Fish 2',
    weight: 400,
    color: 'ORANGE',
    nutritional: {
      calories: 200,
      carbs: 100,
      fat: 100,
      protein: 100,
    },
    category: '3',
    ingredients: [
      {
        id: '1',
        name: 'Bread',
        value: 200,
      },
      {
        id: '2',
        name: 'Meat (Chicken)',
        value: 500,
      },
      {
        id: '3',
        name: 'Cucumber',
        value: 300,
      },
      {
        id: '4',
        name: 'Onion',
        value: 200,
      },
    ],
    imgUrl:
      'https://firebasestorage.googleapis.com/v0/b/react-my-example.appspot.com/o/a2.png?alt=media&token=abe65f60-e23e-48d9-a2c8-13c9e3ccb921',
    favorite: false,
  },
];
