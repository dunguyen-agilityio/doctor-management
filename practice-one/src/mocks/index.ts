export * from './article';
export * from './foods';
export * from './ingredient';
export * from './nutritional';
