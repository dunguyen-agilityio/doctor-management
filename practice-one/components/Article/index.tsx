import { memo } from 'react';
import { Image, StyleSheet, Text, TouchableOpacity, View } from 'react-native';

import { LinearGradient } from 'expo-linear-gradient';

import { ArrowICon, COLORS } from '@constants';

import { IArticle } from '@types';

export interface CardProps {
  title: string;
  image: string;
  type?: CARD_TYPES;
}

type CARD_TYPES = 'green' | 'secondary';

const buttonColor = {
  green: COLORS.GREEN,
  secondary: COLORS.SECONDARY,
};

const Card = (props: IArticle) => {
  const { image, color = 'green', title, backgroundColor } = props;

  return (
    <LinearGradient
      colors={backgroundColor}
      locations={[0, 0.5]}
      start={{ x: 0, y: 0 }}
      end={{ x: 1, y: 1 }}
      style={styles.container}
    >
      <Image
        source={{ uri: image }}
        resizeMode="contain"
        style={{
          height: '80%',
          width: '100%',
          minWidth: 160,
          position: 'absolute',
          right: '-32%',
          bottom: 0,
        }}
      />
      <View style={styles.info}>
        <Text style={[styles.articleText, { color: buttonColor[color] }]}>
          Article
        </Text>
        <Text style={styles.title}>{title}</Text>
        <TouchableOpacity
          style={[styles.button, { backgroundColor: buttonColor[color] }]}
        >
          <Text style={styles.buttonText}>Read now</Text>
          <ArrowICon />
        </TouchableOpacity>
      </View>
    </LinearGradient>
  );
};

export default memo(Card);

const styles = StyleSheet.create({
  articleText: {
    fontSize: 10,
    fontWeight: '600',
    textTransform: 'uppercase',
  },
  title: {
    fontSize: 17,
    fontWeight: '600',
    marginTop: 4,
    fontFamily: 'Signika',
    color: COLORS.LIGHT_BLACK,
  },
  container: {
    height: 169,
    width: 320,
    justifyContent: 'space-between',
    borderRadius: 32,
    resizeMode: 'contain',
    position: 'relative',
    paddingRight: 0,
    overflow: 'hidden',
  },
  info: {
    maxWidth: '60%',
    color: COLORS.LIGHT_BLACK,
    padding: 32,
  },
  button: {
    paddingVertical: 6,
    paddingHorizontal: 16,
    borderRadius: 8,
    marginTop: 8,
    flexDirection: 'row',
    alignItems: 'center',
    width: 104,
  },
  buttonText: {
    color: 'white',
    fontSize: 12,
    fontWeight: '600',
  },
});
