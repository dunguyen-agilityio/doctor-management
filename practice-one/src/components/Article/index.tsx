import { memo } from 'react';
import { Image, StyleSheet, Text, TouchableOpacity, View } from 'react-native';

import { LinearGradient } from 'expo-linear-gradient';

import { ArrowICon, COLORS } from '@/constants';

import { IArticle } from '@/types';

type ARTICLE_TYPES = 'green' | 'secondary';

export interface ArticleProps {
  title: string;
  image: string;
  type?: ARTICLE_TYPES;
}

const buttonColor = {
  green: COLORS.GREEN,
  secondary: COLORS.SECONDARY,
};

const Article = (props: IArticle) => {
  const { image, color, title, backgroundColor } = props;

  return (
    <LinearGradient
      colors={backgroundColor}
      locations={[0, 0.5]}
      start={{ x: 0, y: 0 }}
      end={{ x: 1, y: 1 }}
      style={styles.container}
    >
      <Image
        source={{ uri: image }}
        resizeMode="contain"
        style={{
          height: '80%',
          width: '100%',
          minWidth: 160,
          position: 'absolute',
          right: '-32%',
          bottom: 0,
        }}
      />
      <View style={styles.info}>
        <Text style={[styles.articleText, { color: buttonColor[color] }]}>
          Article
        </Text>
        <Text style={styles.title}>{title}</Text>
        <TouchableOpacity
          style={[styles.button, { backgroundColor: buttonColor[color] }]}
        >
          <Text style={styles.buttonText}>Read now</Text>
          <ArrowICon />
        </TouchableOpacity>
      </View>
    </LinearGradient>
  );
};

export default memo(Article);

const styles = StyleSheet.create({
  articleText: {
    fontSize: 10,
    fontWeight: '600',
    textTransform: 'uppercase',
  },
  title: {
    fontSize: 17,
    fontWeight: '600',
    marginTop: 4,
    fontFamily: 'Signika',
    color: COLORS.LIGHT_BLACK,
  },
  container: {
    height: 169,
    width: 320,
    justifyContent: 'space-between',
    borderRadius: 32,
    resizeMode: 'contain',
    position: 'relative',
    paddingRight: 0,
    overflow: 'hidden',
  },
  info: {
    maxWidth: '60%',
    color: COLORS.LIGHT_BLACK,
    padding: 32,
  },
  button: {
    paddingVertical: 6,
    paddingHorizontal: 16,
    borderRadius: 8,
    marginTop: 8,
    flexDirection: 'row',
    alignItems: 'center',
    width: 104,
  },
  buttonText: {
    color: 'white',
    fontSize: 12,
    fontWeight: '600',
  },
});
