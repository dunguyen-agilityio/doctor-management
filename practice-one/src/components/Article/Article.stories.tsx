import { View } from 'react-native';

import type { Meta, StoryObj } from '@storybook/react';

import { MOCK_ARTICLES } from '@/mocks';

import Article from './index';

const meta = {
  title: 'Article',
  component: Article,
  argTypes: {},
  decorators: [
    (Story) => (
      <View style={{ padding: 16, alignItems: 'flex-start' }}>
        <Story />
      </View>
    ),
  ],
} satisfies Meta<typeof Article>;

export default meta;

type Story = StoryObj<typeof meta>;

export const Default: Story = { args: MOCK_ARTICLES[0] };
