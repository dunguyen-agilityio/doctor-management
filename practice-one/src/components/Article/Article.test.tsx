import React from 'react';

import { render, screen } from '@testing-library/react-native';

import { MOCK_ARTICLES } from '@/mocks';

import Article from './index';

const mockProps = MOCK_ARTICLES[0];
const { name } = mockProps;

describe('Article Component', () => {
  it('renders the title correctly', () => {
    render(<Article {...mockProps} />);
    expect(screen.getByText(name)).toBeTruthy();
  });

  it('renders the article label with correct color', () => {
    const { getByText } = render(<Article {...mockProps} />);
    const articleText = getByText('Article');
    expect(articleText).toBeTruthy();
  });

  it('renders the button with correct text', () => {
    const { getByText } = render(<Article {...mockProps} />);
    expect(getByText('Read now')).toBeTruthy();
  });

  it('matches snapshot', () => {
    const tree = render(<Article {...mockProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
