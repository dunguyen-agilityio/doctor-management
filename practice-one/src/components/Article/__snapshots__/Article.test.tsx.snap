// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article Component matches snapshot 1`] = `
<ViewManagerAdapter_ExpoLinearGradient
  colors={
    [
      4289984168,
      4293656563,
    ]
  }
  endPoint={
    [
      1,
      1,
    ]
  }
  locations={
    [
      0,
      0.5,
    ]
  }
  startPoint={
    [
      0,
      0,
    ]
  }
  style={
    {
      "borderRadius": 32,
      "height": 169,
      "justifyContent": "space-between",
      "overflow": "hidden",
      "paddingRight": 0,
      "position": "relative",
      "resizeMode": "contain",
      "width": 320,
    }
  }
>
  <Image
    resizeMode="contain"
    source={
      {
        "uri": "https://s3-alpha-sig.figma.com/img/8b33/a5b6/24d2e37a7dcbf108d35e8be410c96a58?Expires=1739750400&Key-Pair-Id=A"PKAQ4GOSFWCW27IBOMQ"&Signature=avSEEtN22vSh8aq5Gzzi~bofxqRUv6YwaUlTa0ZvmTH6S-UnlgE4S3sKXqlu~YtbnlHLtmBqMsfniDWO"WRRccMEfs2zvaqOmrx6bTqBQuqMauI8qc93mWyLi0PlDjWuA2OukEGCFPxfF"~3cSSgMkCnxXpkiSUh6UvXQCHUGrb3smseU6TKWtbBp3MHkG5pCP8NKockzv0FBZQvuLgITaGQ9mFpD3Th5xHP0qpGfG6HUrDD6bSnKTLopvWrJ~8XyYweSy7yi6yiwEsT-KeC2zdXQGy5mMAGc0VuVPQeUcbmesl9SE4qGiNx7kTe59Qg41NHJ-mFj1aVEhu50Amx6m8g__",
      }
    }
    style={
      {
        "bottom": 0,
        "height": "80%",
        "minWidth": 160,
        "position": "absolute",
        "right": "-32%",
        "width": "100%",
      }
    }
  />
  <View
    style={
      {
        "color": "#330600",
        "maxWidth": "60%",
        "padding": 32,
      }
    }
  >
    <Text
      style={
        [
          {
            "fontSize": 10,
            "fontWeight": "600",
            "textTransform": "uppercase",
          },
          {
            "color": "#27AE60",
          },
        ]
      }
    >
      Article
    </Text>
    <Text
      style={
        {
          "color": "#330600",
          "fontFamily": "Signika",
          "fontSize": 17,
          "fontWeight": "600",
          "marginTop": 4,
        }
      }
    >
      The pros and cons of fast food.
    </Text>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#27AE60",
          "borderRadius": 8,
          "flexDirection": "row",
          "marginTop": 8,
          "opacity": 1,
          "paddingHorizontal": 16,
          "paddingVertical": 6,
          "width": 104,
        }
      }
    >
      <Text
        style={
          {
            "color": "white",
            "fontSize": 12,
            "fontWeight": "600",
          }
        }
      >
        Read now
      </Text>
      <Image
        source={
          {
            "testUri": "../../../assets/icons/arrow.png",
          }
        }
      />
    </View>
  </View>
</ViewManagerAdapter_ExpoLinearGradient>
`;
